notifications_utils/markdown.py

```diff
@@ -23,45 +23,37 @@
     r"^( *)([•*-]|\d+\.)[\s\S]+?"
     r"(?:"
     r"\n+(?=\1?(?:[-*_] *){3,}(?:\n+|$))"  # hrule
-    r"|\n+(?=%s)"  # def links
-    r"|\n+(?=%s)"  # def footnotes
+    rf"|\n+(?={mistune._pure_pattern(mistune.BlockGrammar.def_links)})"  # def links
+    rf"|\n+(?={mistune._pure_pattern(mistune.BlockGrammar.def_footnotes)})"  # def footnotes
     r"|\n{2,}"
     r"(?! )"
     r"(?!\1(?:[•*-]|\d+\.) )\n*"
     r"|"
     r"\s*$)"
-    % (
-        mistune._pure_pattern(mistune.BlockGrammar.def_links),
-        mistune._pure_pattern(mistune.BlockGrammar.def_footnotes),
-    )
 )
 mistune.BlockGrammar.list_item = re.compile(
-    r"^(( *)(?:[•*-]|\d+\.)[^\n]*" r"(?:\n(?!\2(?:[•*-]|\d+\.))[^\n]*)*)", flags=re.M
+    r"^(( *)(?:[•*-]|\d+\.)[^\n]*(?:\n(?!\2(?:[•*-]|\d+\.))[^\n]*)*)", flags=re.M
```

Comment:
i dont think we need to change anything now but i wonder if this was originally a multiline string that got one-lined, and now smooshed together, and if we think it's more readable over multiple lines?

```py
mistune.BlockGrammar.list_item = re.compile(
  (
    r"^(( *)(?:[•*-]|\d+\.)[^\n]*"
    r"(?:\n(?!\2(?:[•*-]|\d+\.))[^\n]*)*)"
  ),
  flags=re.M
)
```

Reply:
Black re-smooshes it if you try to do this 😞